### PR TITLE
Make resize cursor stay while dragging

### PR DESCRIPTION
**Pull request by xyzqm**
Release Notes:

- Resize cursor stays while dragging (it previously would switch back to a normal arrow cursor)

Before:
https://github.com/user-attachments/assets/e69fa5d5-fe95-4233-8d83-89b51b41060f

After:

https://github.com/user-attachments/assets/6a161eca-335f-4a3b-ba6b-58ac58d4174d

### CODE REVIEW

**Comment by danielzsh** [comment]
Currently the cursor still switches back to arrow half the time and I'm not too sure why...

**Comment by JunkuiZhang** [comment]
This PR might not be accepted by the Zed team, see #11596.

File: crates/gpui/src/window.rs [hunk 1]
```diff
@@ -3060,11 +3062,13 @@ impl<'a> WindowContext<'a> {
             }
             PlatformInput::MouseDown(mouse_down) => {
                 self.window.mouse_position = mouse_down.position;
+                self.window.mouse_down = true;
```
Comment (mikayla-maki) [inline crates/gpui/src/window.rs:3065]: I don't think this solution is sufficient yet, as the mouse exiting the window, releasing, and then re-entering might cause this `mouse_down` state to desync from the real mouse state. Maybe set it on mouse move too? We might need other events, let me know what you find :)

**Review by mikayla-maki — CHANGES_REQUESTED** [review]

**Comment by mikayla-maki** [comment]
Since it's been a month since I requested changes to this, I'm going to close this PR for now. But we'd still like this bug fix :)

**Pull request closed without merging**